change/@ni-nimble-components-51de64d0-1537-4dce-89b5-e85857542ebb.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Scrollable tabs for Tabs and AnchorTabs.",
```

Comment:
The problem lies in the lack of the `ResizeObserver` firing when changing from "Tab Two" to "2" (so tabs are now "1", "2", and "Tab Three") . The reason for this is that the content rect for the `tablist` hasn't changed yet, because it has already been reduced because of the presence of the buttons.

Fixing this, I believe, would require a more heavy-handed approach using a `MutationObserver` on the content of each tab (note that I don't think an `IntersectionObserver` would resolve this either). At the moment, I'm not convinced that this is a big enough issue to resolve as this kind of asymmetric behavior isn't that bad.

Reply:
I agree that this is a pretty low-impact issue that isn't very likely to occur. I'm fine not fixing it if the solution would be overly complex or inefficient.

Reply:
I didn't fully follow the discussion but just to confirm we aren't ending up in a scenario where sometimes the tabs scroll left and right are not visible when they are needed right? It's only that sometimes they are still show when they could instead be hidden?

Reply:
Correct, there are scenarios where the scroll buttons aren't necessary but are shown anyway. There is no known scenario where we should be showing the buttons but do not.